services/user/XAdmin/ui/src/pages/keys-page.tsx

```diff
@@ -0,0 +1,517 @@
+                                                ) : (
+                                                    <Lock
+                                                        className="h-4 w-4"
+                                                        color="red"
+                                                    />
+                                                )}
+                                            </TableCell>
+                                        </TableRow>
+                                    ))}
+                                </TableBody>
+                            </Table>
+                        </div>
+                    </TabsContent>
+                    <TabsContent value="keys">
+                        {serverKeys && serverKeys?.length === 0 ? (
+                            <div className="mt-4 w-full">
+                                <EmptyBlock
+                                    buttonLabel={
+                                        keyDevices.every((kd) => !kd.unlocked)
+                                            ? undefined
+                                            : "New Key"
+                                    }
+                                    onClick={() => setNewKeyDialogOpen(true)}
+                                    title="No server keys available"
+                                    description={
+                                        keyDevices.every((kd) => !kd.unlocked)
+                                            ? "No server keys are available. Unlock a device to view keys in that device or create a new key to continue."
+                                            : "There are no server keys available to the server for signing."
+                                    }
+                                    ButtonIcon={FileKey2}
+                                />
+                            </div>
+                        ) : (
+                            <div className="rounded-md border">
+                                <Table className="table-fixed">
+                                    <TableHeader>
+                                        <TableRow>
+                                            <TableHead className="w-32 text-center">
+                                                Auth service
+                                            </TableHead>
+                                            <TableHead className="w-full">
+                                                Public key (hex-encoded DER)
+                                            </TableHead>
+                                            <TableHead className="w-16 text-right"></TableHead>
+                                        </TableRow>
+                                    </TableHeader>
+                                    <TableBody>
+                                        {serverKeys?.map((key) => (
+                                            <TableRow key={key.rawData}>
```

Comment:
I think it would be better to show the key fingerprint instead of the raw hex of the key.

Reply:
🙏 That's much better. Take a look and let me know if that's okay, please.

Reply:
lgtm!